components/load_from_files/src/main.py

```diff
@@ -0,0 +1,336 @@
+
+    Methods:
+    -------
+        to_dask_dataframe():
+            Converts the read file content to binary form and returns it as a Dask DataFrame.
+    """
+
+    def __init__(self, handler):
+        """
+        Constructs all the necessary attributes for the file converter object.
+
+        Parameters
+        ----------
+            handler: handles the files and reads their content.
+                     Should have a 'read' method that yields tuples (file_name, file_content).
+        """
+        self.handler = handler
+
+    @staticmethod
+    @delayed
+    def create_record(file_name, file_content):
+        """
+        Static helper method that creates a dictionary record of file name and its content in
+          string format.
+
+        Parameters
+        ----------
+        file_name : str
+            Name of the file.
+        file_content : str
+            The content of the file.
+
+        Returns:
+        -------
+        dict
+            A pandas dataframe with 'filename' as index and 'Content' column for binary file
+            content.
+        """
+        record = pd.DataFrame(data={"filename": file_name, "Content": file_content})
+        # convert 'Content' column to 'object' type if it's not already
+        if record["Content"].dtype != "object":
+            record["Content"] = record["Content"].astype("str")
+        return record
+
+    def to_dask_dataframe(self):
+        """
+        This method converts the read file content to binary form and returns a Dask DataFrame.
+
+        Returns:
+        ------
+        dask.dataframe
+            The created Dask DataFrame with filenames as indices and file content in binary form
+            as Content column.
+        """
+        # Initialize an empty list to hold all our records in 'delayed' objects.
+        # These objects encapsulate our function calls and their results, but don't execute
+        # just yet.
+        records = []
+        logging.debug("Converting individual files into a dask dataframe ......")
+        # Iterate over each file handled by the handler
+        for file_name, file_content in self.handler.read():
+            # Create a record for each file - note that this doesn't really run just yet due to
+            # being 'delayed'
+            record = self.create_record(file_name, file_content)
+            # Add the 'delayed' record to our list
+            records.append(record)
+
+        #  Create an empty pandas dataframe with correct column names and types as meta.
+        metadata = pd.DataFrame(
+            data={
+                "filename": pd.Series([], dtype="object"),
+                "Content": pd.Series([], dtype="bytes"),
+            },
+        )
+
+        # Use the delayed objects to create a Dask DataFrame. Dask knows how to handle these and
+        # will efficiently compute them when needed.
+        dataframe = dd.from_delayed(records, meta=metadata)
```

Comment:
I think this creates a partition for each record ([docs](url)). So we should probably repartition the resulting dataframe.

CC: @PhilippeMoussalli

Reply:
Oh is it, Ill look into it and refractor it